Storage/Bookmarks/BookmarksModel.swift

```diff
@@ -134,6 +134,7 @@ public protocol BookmarksModelFactory {
     var nullModel: BookmarksModel { get }
 
     func isBookmarked(url: String) -> Deferred<Maybe<Bool>>
+    func getURLForKeywordSearch(keyword: String) -> Deferred<Maybe<String>>
```

Comment:
I think this should not be on `BookmarksModel`. Instead, put it on a new protocol — perhaps `KeywordSearchSource` — to which `SQLiteBookmarks` conforms and which `Profile.bookmarks` requires.